internal/scan-manager/core/src/main/java/org/wso2/security/tools/scanmanager/core/service/ScanEngineServiceImpl.java

```diff
@@ -266,47 +263,126 @@ private void initiateScanRequest(Scan scan, ScannerApp scannerApp) throws ScanMa
             }
         } catch (RestClientException e) {
             throw new ScanManagerException("Error occurred while connecting to the scanner service endpoint");
         }
     }
 
-    private Container createContainer(Scan scan, ScannerApp scannerApp, String containerHost,
-                                      Integer containerPort) throws ScanManagerException {
+    private Container createContainer(Scan scan, ScannerApp scannerApp, String containerHost, Integer containerPort)
+            throws ScanManagerException {
         Map<String, String> labels = new HashMap<>();
         labels.put(CONTAINER_SCAN_JOB_ID_LABEL_NAME, scan.getJobId());
         labels.put(CONTAINER_APP_LABEL_NAME, scannerApp.getAppId());
         labels.put(CONTAINER_SCANNER_LABEL_NAME, scannerApp.getScanner().getName());
 
-        String[] envVariables =
-                new String[]{CONTAINER_ENV_NAME_SCAN_MANAGER_HOST + "=" + ScanManagerConfiguration.getInstance()
-                        .getScanManagerHost(), CONTAINER_ENV_NAME_SCAN_MANAGER_PORT + "=" + ScanManagerConfiguration
-                        .getInstance().getScanManagerPort()};
-        return dockerHandler.create(scannerApp.getScanner().getImage(), containerHost, containerPort, labels,
-                new ArrayList<>(), envVariables);
+        String[] envVariables = new String[] { CONTAINER_ENV_NAME_SCAN_MANAGER_HOST + "=" +
+                ScanManagerConfiguration.getInstance()
+                        .getScanManagerHost(),
+                CONTAINER_ENV_NAME_SCAN_MANAGER_PORT + "=" + ScanManagerConfiguration.getInstance()
+                        .getScanManagerPort() };
+        return dockerHandler
+                .create(scannerApp.getScanner().getImage(), containerHost, containerPort, labels, new ArrayList<>(),
+                        envVariables);
+    }
+
+    /**
+     * This scheduler service will be executed during the scan manager start up. This will periodically check
+     * whether container is not exist for RUNNING or SUBMITTED scans, if docker container is not exist container
+     * will be restarted.
+     *
+     * @param schduler ScheduledExecutorService
+     */
+    private void activateScanResilienceTask(ScheduledExecutorService schduler) {
+        Runnable checkContainerStatusTask = new CheckContainerStatusTask();
+        schduler.scheduleWithFixedDelay(checkContainerStatusTask, INITIAL_DELAY, DELAY_BETWEEN_RUNS, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Perform scan recovery task upon availability of container.
+     *
+     * @param containerInfo Container information
+     * @param scan          Scan object
+     */
+    private void doScanRecovery(Container containerInfo, Scan scan) {
+
+        // If container is not exist, restart the container
+        try {
+            dockerHandler.restart(containerInfo.getId());
+
+            // Sleep till the scanner service is started.
+            Thread.sleep(SCANNER_SERVICE_WAIT_TIME);
```

Comment:
Shouldn't we implement a health  check here. It's not a guarantee that the service will be up within `SCANNER_SERVICE_WAIT_TIME`

Reply:
As per discussed, https://github.com/wso2-enterprise/security-artifacts/issues/451 created and updated new issue to look this separately.